engine/src/org/pentaho/di/job/entries/zipfile/JobEntryZipFile.java

```diff
@@ -977,9 +970,15 @@ public Result execute( Result previousResult, int nr ) {
     return result;
   }
 
+  private Result errorResult( Result result ) {
+    result.setNrErrors( 1 );
```

Comment:
Shouldn't this line be:
result.increaseErrors( 1 );
Otherwise, it will only ever report one error, even if there are multiple rows with errors.